### PR TITLE
fix: initialize auto suggestions providers (closes #571)

### LINKED ISSUE #571: bug: Using Copilot for Auto Suggestions
### Describe the bug

Editing a file after enabling suggestions I get the following error every time it tries to suggest:
`^I...al/share/nvim/lazy/avante.nvim/lua/avante/utils/init.lua:507: in function <...al/share/nvim/lazy/avante.nvim/lua/avante/utils/init.lua:506>`

Seems like this error is related to the `suggest()` function, as I'm unfamiliar with the code and architecture it's hard for me to pinpoint more specifically where the error originates more than this.
I've install Copilot using LazyExtras, if I don't do this I get an error from the avante suggestions implying that the `copilot.setup` function was not called, this might be a related issue with the fact that I'm using LazyVim (probably).
Love the Avante project BTW as someone who doesn't want to leave neovim just because of some simple but powerful AI helper tools :)

### To reproduce

Using lazyvim I have the following 'avante.lua' file inside my plugins folder:
```lua
return {
  "yetone/avante.nvim",
  event = "VeryLazy",
  lazy = false,
  version = false, -- set this if you want to always pull the latest change
  -- build = "powershell -ExecutionPolicy Bypass -File Build.ps1 -BuildFromSource false" -- for windows
  dependencies = {
    "stevearc/dressing.nvim",
    "nvim-lua/plenary.nvim",
    "MunifTanjim/nui.nvim",
    --- The below dependencies are optional,
    "nvim-tree/nvim-web-devicons", -- or echasnovski/mini.icons
    "zbirenbaum/copilot.lua",
    {
      -- support for image pasting
      "HakonHarnes/img-clip.nvim",
      event = "VeryLazy",
      opts = {
        -- recommended settings
        default = {
          embed_image_as_base64 = false,
          prompt_for_file_name = false,
          drag_and_drop = {
            insert_mode = true,
          },
          -- required for Windows users
          use_absolute_path = true,
        },
      },
    },
    {
      -- Make sure to set this up properly if you have lazy=true
      "MeanderingProgrammer/render-markdown.nvim",
      opts = {
        file_types = { "markdown", "Avante" },
      },
      ft = { "markdown", "Avante" },
    },
  },
  opts = {
    -- add any opts here
  },
  init = function()
    local config = require("avante.config")
    config.auto_suggestions_provider = "copilot"
    -- vim.notify("Avante initialization complete", vim.log.levels.INFO)
  end,
  -- if you want to build from source then do `make BUILD_FROM_SOURCE=true`
  build = "make",
}
```
Pretty much similar to the README file.

### Expected behavior

Suggestions will work seamlessly.

### Environment

- Arch Linux (Linux 6.10.9-arch1-1)
- Neovim v0.10.1
- LazyVim

### PR DESCRIPTION
**Pull request by holaguz**
Fixes #571

### CODE REVIEW

**Comment by LessComplexity** [comment]
Did the fix locally and it worked :)
Waiting for an official update ;)
thanks

**Comment by holaguz** [comment]
It seems to be working. 🤔 

Autocompletion with OpenAI provider (also, the settings I'm currently using):

image

```js
{
  body = {
    max_tokens = 4096,
    model = "gpt-4o-mini",
    stream = true,
    temperature = 0
  },
  url = "https://api.openai.com/v1/chat/completions"
}
```

Also a chat with Claude:

image

```js
{
  body = {
    max_tokens = 4096,
    model = "claude-3-haiku-20240307",
    stream = true,
    system = { {
        cache_control = {
          type = "ephemeral"
        },
        text = "You are an excellent programming expert.\n",
        type = "text"
      } },
    temperature = 0
  },
  url = "https://api.anthropic.com/v1/messages"
}
```

**Review by aarnphm — APPROVED** [review]

File: lua/avante/providers/init.lua [hunk 1]
```diff
@@ -302,7 +302,12 @@ M.setup = function()
 
   ---@type AvanteProviderFunctor
   local provider = M[Config.provider]
+  local auto_suggestions_provider = M[Config.auto_suggestions_provider]
   E.setup({ provider = provider })
+
+  if auto_suggestions_provider and auto_suggestions_provider ~= provider then
+    E.setup({ provider = auto_suggestions_provider })
```
Comment (yetone) [inline lua/avante/providers/init.lua:309]: The design patterns related to the provider are a bit too complex. Why is `E.setup` called twice here? @aarnphm

Comment (aarnphm) [reply lua/avante/providers/init.lua:309]: Ye I'm intending to clean this up a bit. It has to do with we are lazily loading the env for each providers.

**Pull request merged**